Zend/zend_API.h

```diff
@@ -440,6 +440,7 @@ ZEND_API void zend_declare_class_constant_double(zend_class_entry *ce, const cha
 ZEND_API void zend_declare_class_constant_stringl(zend_class_entry *ce, const char *name, size_t name_length, const char *value, size_t value_length);
 ZEND_API void zend_declare_class_constant_string(zend_class_entry *ce, const char *name, size_t name_length, const char *value);
 
+zend_result zend_update_class_constant(zend_class_constant *c, zval *value, zend_class_entry *scope);
```

Comment:
This needs a `ZEND_API` to be available from dynamic libraries. Thus your linker error.